libfreerdp/core/proxy.c

```diff
@@ -253,3 +257,136 @@ BOOL http_proxy_connect(BIO* bufferedBio, const char* hostname, UINT16 port)
 	return TRUE;
 }
 
+/* SOCKS Proxy auth methods by rfc1928 */
+#define AUTH_M_NO_AUTH   0
+#define AUTH_M_GSSAPI    1
+#define AUTH_M_USR_PASS  2
+
+static BOOL socks_proxy_connect(BIO* bufferedBio, const char* hostname, UINT16 port)
+{
+	int status;
+	BYTE buf[280], hostnlen = 0xff & strlen(hostname);
+	/* CONN REQ replies in enum. order */
+	static const char *rplstat[] = {
+	  "succeeded",
+	  "general SOCKS server failure",
+	  "connection not allowed by ruleset",
+	  "Network unreachable",
+	  "Host unreachable",
+	  "Connection refused",
+	  "TTL expired",
+	  "Command not supported",
+	  "Address type not supported"
+	};
+
+	/* select auth. method */
+	memset(buf, '\0', sizeof(buf));
```

Comment:
this memset is not needed

Reply:
I apologize but I was busy on garden during the weekend.
buf[] is the "automatic variable" where the initial value can not be assumed so for portability reason it is needed. However the memset() here is performed once/per session. What do you think it will "save" when it will not be here?

Reply:
...or did you meant line 295? memset() can be removed from there but if you debug (might happen) then it is also better to zero a buffer before use. "Costs" ale low but usefulness is high.